src/qibolab/instruments/qblox/cluster_qcm_bb.py

```diff
@@ -252,17 +252,12 @@ def _get_next_sequencer(self, port, frequency, qubits: dict):
         Raises:
             Exception = If attempting to set a parameter without a connection to the instrument.
         """
-        # select the qubit relative to specific port
+        # select the qubit with flux line, if present, connected to the specific port
         qubit = None
         for _qubit in qubits.values():
-            if _qubit.flux.port is not None:
-                if (
-                    _qubit.flux.port.name == port
-                    and _qubit.flux.port.module.name == self.name
-                ):
-                    qubit = _qubit
-            else:
-                log.warning(f"Qubit {_qubit.name} has no flux line connected")
+            if _qubit.flux and _qubit.flux.port == self.ports[port]:
```

Comment:
```suggestion
            if _qubit.flux is not None and _qubit.flux.port == self.ports[port]:
```
Don't use the bare truthiness of the attribute, it is error-prone.

Reply:
Thanks!